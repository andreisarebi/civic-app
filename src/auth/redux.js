export const AUTH_NAMESPACE = 'auth';
export const formTypes = {
  INITIAL: 'initial',
  LOGIN: 'login',
  SIGN_UP: 'signUp',
};

// Actions
export const AuthActionType = {
  EmailLoginRequest: 'civicApp/auth/emailLoginRequest',
  RegisterRequest: 'civicApp/auth/registerRequest',
  FacebookLoginRequest: 'civicApp/auth/facebookLoginRequest',
  GoogleLoginRequest: 'civicApp/auth/googleLoginRequest',
  LoginSuccess: 'civicApp/auth/emailLoginSuccess',
  LogOutRequest: 'civicApp/auth/logOutRequest',
  LogOutSuccess: 'civicApp/auth/logOutSuccess',
  AuthFailure: 'civicApp/auth/failure',
  SwitchFormType: 'civicApp/auth/switchFormType',
  UpdateEmail: 'civicApp/auth/updateEmail',
  UpdatePassword: 'civicApp/auth/updatePassword',
  UpdateDuplicatePassword: 'civicApp/auth/updateDuplicatePassword',
  ShowErrors: 'civicApp/auth/showErrors',
};

// Action Creators
export const emailLogin = () => ({ type: AuthActionType.EmailLoginRequest });

export const register = () => ({ type: AuthActionType.RegisterRequest });

export const facebookLogin = () => ({ type: AuthActionType.FacebookLoginRequest });

export const googleLogin = () => ({ type: AuthActionType.GoogleLoginRequest });

// Action Creators
export const logIn = (email, password) => ({
  type: AuthActionType.LoginRequest,
  payload: { email, password },
});

export const loginSuccess = user => ({
  type: AuthActionType.LoginSuccess,
  payload: user,
});

export const logOut = () => ({ type: AuthActionType.LogOutRequest });

export const logOutSuccess = () => ({ type: AuthActionType.LogOutSuccess });

export const authFailure = error => ({
  type: AuthActionType.AuthFailure,
  payload: error,
});

/*
 * @param formType: member of formTypes object;
 */
export const switchFormType = formType => ({
  type: AuthActionType.SwitchFormType,
  payload: formType,
});

/*
 * @param email: string;
 */
export const updateEmail = email => ({
  type: AuthActionType.UpdateEmail,
  payload: email,
});

/*
 * @param password: string;
 */
export const updatePassword = password => ({
  type: AuthActionType.UpdatePassword,
  payload: password,
});

/*
 * @param password: string;
 */
export const updateDuplicatePassword = password => ({
  type: AuthActionType.UpdateDuplicatePassword,
  payload: password,
});

/*
 * @param shouldShow: boolean;
 */
export const showErrors = shouldShow => ({
  type: AuthActionType.ShowErrors,
  payload: shouldShow,
});

/*
 * When a user is logged in, we will store:
 * id, email, ...
 */
export const initialState = {
  user: null,
  formType: formTypes.INITIAL,
  email: '',
  password: '',
  duplicatePassword: '',
  showErrors: false,
};

const authReducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case AuthActionType.SwitchFormType:
      return {
        ...state,
        formType: action.payload,
        showErrors: false,
      };
    case AuthActionType.UpdateEmail:
      return {
        ...state,
        email: action.payload,
        showErrors: false,
      };
    case AuthActionType.UpdatePassword:
      return {
        ...state,
        password: action.payload,
        showErrors: false,
      };
    case AuthActionType.UpdateDuplicatePassword:
      return {
        ...state,
        duplicatePassword: action.payload,
        showErrors: false,
      };
    case AuthActionType.ShowErrors:
      return {
        ...state,
        showErrors: action.payload,
      };
    case AuthActionType.LoginSuccess:
      return {
        ...state,
        user: { id: action.payload.uid, email: action.payload.email },
      };
    case AuthActionType.LogOutSuccess:
      return {
        ...state,
        user: null,
      };
    default:
      return state;
  }
};

export default authReducer;
