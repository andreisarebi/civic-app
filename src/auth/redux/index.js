import { combineReducers } from 'redux';
import userReducer, { USER_NAMESPACE } from './userReducer';
import formReducer, { FORM_NAMESPACE } from './formReducer';

export const AUTH_NAMESPACE = 'auth';

const authReducer = combineReducers({
  [USER_NAMESPACE]: userReducer,
  [FORM_NAMESPACE]: formReducer,
});

export default authReducer;
