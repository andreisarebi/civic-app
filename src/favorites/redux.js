import { Category } from './models';
import { UserActionType } from '../user/redux';

// Selectors
export const getUserFavorites = (state) => ({
  [Category.Candidates]: Array.from(getFavoritesForCategory(state, Category.Candidates)),
  [Category.Events]: Array.from(getFavoritesForCategory(state, Category.Events)),
});

export const getIsFavorite = (state, id, category) => (
  getFavoritesForCategory(state, category).has(id)
);

export const getFavoritesForCategory = (state, category) => (
  state[FAVORITES_NAMESPACE][category]
);

export const FAVORITES_NAMESPACE = 'favorites';

// Action Creators
export const favoritesRequestSuccess = favorites => ({
  type: FavoritesActionType.RequestSuccess,
  payload: favorites,
});

export const toggleFavorite = (id, category) => ({
  type: FavoritesActionType.Toggle,
  payload: { id, category },
});

export const addOrRemoveFavorite = (id, category, shouldAdd) => (
  shouldAdd ? addFavorite(id, category) : removeFavorite(id, category)
);

const addFavorite = (id, category) => ({
  type: FavoritesActionType.Add,
  payload: { id, category }
});

const removeFavorite = (id, category) => ({
  type: FavoritesActionType.Remove,
  payload: { id, category },
});

export const FavoritesActionType = {
  Toggle: 'civicApp/favorites/toggle',
  Add: 'civicApp/favorites/add',
  Remove: 'civicApp/favorites/remove',
};

// Reducer

const reducer = (state = initialState, action) => {
  let newState;
  switch (action.type) {
    case UserActionType.RequestSuccess:
      return action.payload.favorites ? {
        [Category.Candidates]: new Set(action.payload.favorites[Category.Candidates]),
        [Category.Events]: new Set(action.payload.favorites[Category.Events]),
      } : state;
    case FavoritesActionType.Add:
      return {
        ...state,
        [action.payload.category]: (new Set(state[action.payload.category])).add(action.payload.id),
      };
    case FavoritesActionType.Remove:
      newState = new Set(state[action.payload.category]);
      newState.delete(action.payload.id);
      return {
        ...state,
        [action.payload.category]: newState,
      };
    default:
      return state;
  }
};

const initialState = {
  [Category.Candidates]: new Set(),
  [Category.Events]: new Set(),
};

export default reducer;
