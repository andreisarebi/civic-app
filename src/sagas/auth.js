import { call, takeEvery, put } from 'redux-saga/effects';
import {
  subscribeToAuthStateChanges,
  logIn,
  logOut,
  register,
} from '../auth/api';
import { AuthActionType, logOutSuccess, loginSuccess } from './redux';

const authSaga = function*() {
  yield call(subscribeToAuthStateChanges, loginSuccess, logOutSuccess); // doesn't need a yield but makes testing easier
  yield takeEvery(AuthActionType.LoginRequest, loginSaga);
  yield takeEvery(AuthActionType.LogOutRequest, logOutSaga);
  yield takeEvery(AuthActionType.RegisterRequest, registerSaga);
};

export const loginSaga = function*(action) {
  try {
    const user = yield call(logIn, action.payload);
    yield put(loginSuccess(user));
  } catch (err) {
    // handle error
  }
};

export const logOutSaga = function*() {
  try {
    yield call(logOut);
    yield put(logOutSuccess());
  } catch (err) {
    // handle error
  }
};

export const registerSaga = function*(action) {
  try {
    yield call(register, action.payload);
    yield call(loginSaga, action);
  } catch (err) {
    // handle error
  }
};

export default authSaga;
