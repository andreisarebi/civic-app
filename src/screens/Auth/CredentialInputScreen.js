import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, View, Text } from 'react-native';
import EmailInput from './EmailInput';
import PasswordInput from './PasswordInput';
import FacebookAuthButton from './FacebookAuthButton';
import GoogleAuthButton from './GoogleAuthButton';
import colors from '../../styles/colors';
import { formTypes } from '../../auth/redux';

class CredentialInputScreen extends React.Component {
  static propTypes = {
    isLoggedIn: PropTypes.bool.isRequired,
    navigation: PropTypes.shape({ navigate: PropTypes.func }).isRequired,
    onSubmit: PropTypes.func.isRequired,
    formType: PropTypes.string.isRequired,
    changeFormType: PropTypes.func.isRequired,
    formIsValid: PropTypes.bool.isRequired,
    email: PropTypes.string.isRequired,
    updateEmail: PropTypes.func.isRequired,
    password: PropTypes.string.isRequired,
    updatePassword: PropTypes.func.isRequired,
    isLoading: PropTypes.bool.isRequired,
    errorMessage: PropTypes.string.isRequired,
    updateErrorVisibility: PropTypes.func.isRequired,
    showErrors: PropTypes.bool.isRequired,
    duplicatePassword: PropTypes.string,
    updateDuplicatePassword: PropTypes.func,
  };

  static defaultProps = {
    duplicatePassword: '',
    updateDuplicatePassword: () => undefined,
  };

  componentDidUpdate() {
    if (this.props.isLoggedIn) {
      this.props.navigation.navigate('App');
    }
  }

  getOptions = (formType) => {
    switch (formType) {
      case formTypes.LOGIN:
        return {
          titleText: 'Sign in',
          changeFormText: "Don't have an account yet? Register",
          otherFormType: formTypes.SIGN_UP,
          googleButtonText: 'Sign in with Google',
        };
      case formTypes.SIGN_UP:
        return {
          titleText: 'Sign Up for Civic',
          changeFormText: 'Have an account? Sign in',
          otherFormType: formTypes.LOGIN,
          googleButtonText: 'Sign up with Google',
        };
      default:
        return {};
    }
  };

  handleSubmit = () => {
    if (!this.props.isLoading) {
      if (this.props.formIsValid) {
        this.props.onSubmit();
      } else {
        this.props.updateErrorVisibility(true);
      }
    }
  };

  render() {
    const options = this.getOptions(this.props.formType);
    return (
      <View style={styles.container}>
        <View style={styles.inputContainer}>
          <Text style={styles.titleText}>{options.titleText}</Text>
          <EmailInput value={this.props.email} onChangeText={this.props.updateEmail} />
          <PasswordInput value={this.props.password} onChangeText={this.props.updatePassword} />
          {this.props.formType === formTypes.SIGN_UP && (
            <PasswordInput
              value={this.props.duplicatePassword}
              onChangeText={this.props.updateDuplicatePassword}
              placeholder="Re-type password"
            />
          )}
          {this.props.showErrors && <Text style={styles.errorMessage}>{this.props.errorMessage}</Text>}
          <Text onPress={this.handleSubmit} style={styles.submitButton}>
            SUBMIT
          </Text>
          <Text style={styles.text}>or</Text>
          <GoogleAuthButton title={options.googleButtonText} style={styles.social} />
          <FacebookAuthButton title="Continue with Facebook" style={styles.social} />
          <Text onPress={() => this.props.changeFormType(options.otherFormType)} style={styles.text}>
            {options.changeFormText}
          </Text>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    backgroundColor: colors.lightBlue,
    paddingTop: 10,
    paddingBottom: 10,
  },
  inputContainer: {
    backgroundColor: colors.white,
    borderTopLeftRadius: 25,
    borderTopRightRadius: 25,
    borderBottomLeftRadius: 25,
    borderBottomRightRadius: 25,
    flexDirection: 'column',
    justifyContent: 'center',
    marginLeft: 10,
    marginRight: 10,
    maxHeight: 507,
    maxWidth: 369,
    paddingTop: 20,
    paddingBottom: 20,
    paddingLeft: 20,
    paddingRight: 20,
  },
  social: {
    marginBottom: 8,
    marginLeft: 0,
    marginRight: 0,
    alignSelf: 'center',
    width: '100%',
    maxWidth: 322,
  },
  submitButton: {
    textAlign: 'right',
    paddingTop: 20,
    fontSize: 18,
    color: colors.darkBlue,
  },
  titleText: {
    textAlign: 'left',
    fontSize: 24,
    fontWeight: '500',
    marginBottom: 20,
  },
  text: {
    textAlign: 'center',
    fontSize: 18,
    paddingTop: 10,
    paddingBottom: 20,
  },
  errorMessage: {
    color: colors.red,
    fontSize: 16,
    textAlign: 'center',
    paddingTop: 4,
    paddingBottom: 4,
  },
});

export default CredentialInputScreen;
