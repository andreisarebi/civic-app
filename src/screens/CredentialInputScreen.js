import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux'
import { StyleSheet, View, Text, TextInput } from 'react-native';
import colors from '../styles/colors';
import SocialButton from '../auth/SocialButton';
import firebase, { auth, provider } from '../firebase/initialize';
import Expo from 'expo';
import { CredentialInput } from '../welcome/LogIn';
import { getIsLoggedIn, register, loginSuccess, logOutSuccess } from '../auth/redux'
//import { signInWithGoogleAsync, signInWithFacebookAsync } from '../auth/socialauth'

class CredentialInputScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      duplicatePassword: '',
      user: null
    };
  }

  handleSubmit = () => {
    this.props.onSubmit(this.state.email, this.state.password);
  }

  checkPassword() {
    return this.state.password === this.state.duplicatePassword;
  }

  componentDidMount() {
      auth.onAuthStateChanged((user) => {
          if (user) {
              this.setState({ user });
              this.props.onLogIn(user);
          }
      });
  }

  logout = () => {
      auth.signOut()
          .then(() => {
              this.setState({
                  user: null
              });
              this.props.onLogOut();
          });
  }

  gotToSurvey() {
    this.props.navigation.navigate('Survey');
  }

  async signUpWithGoogleAsync() {
      try {
          const result = await Expo.Google.logInAsync({
              androidClientId: '506898842953-a5djvc12er7cbmv78ajfjidokjmlropn.apps.googleusercontent.com',
              iosClientId: '506898842953-8nise7b8pq8ifdp9qpjta6d5no0l5u93.apps.googleusercontent.com',
              scopes: ['profile', 'email'],
          });

          if (result.type === 'success') {
              this.props.onLogIn(result);
              return result.accessToken;
          } else {
              return { cancelled: true };
          }
      } catch (e) {
          return { error: true };
      }
  }

  async signUpWithFacebookAsync() {
      const { type, token } = await Expo.Facebook.logInWithReadPermissionsAsync('206331633410454', {
          permissions: ['public_profile'],
      });
      if (type === 'success') {
          // Build Firebase credential with the Facebook access token.
          const credential = firebase.auth.FacebookAuthProvider.credential(token);

          // Sign in with credential from the Facebook user.
          firebase.auth().signInWithCredential(credential).catch((error) => {
              // Handle Errors here.
          });
      }
  }

  render() {
    if (this.props.isLoggedIn) {
      this.gotToSurvey();
    }
      return (
          <View style={styles.container}>
              <View style={styles.inputContainer}>
                  <Text style={styles.titleText}>Sign Up for Civic</Text>
                  <CredentialInput onSubmit={this.props.register}/>
                  <Text style={styles.text}>or</Text>
                  <SocialButton
                      type="google"
                      title="Sign up with Google"
                      style={styles.social}
                      onPress={this.signUpWithGoogleAsync}
                      onLongPress={this.signUpWithGoogleAsync} />
                  <SocialButton type="facebook" title="Continue with Facebook" style={styles.social}
                      onPress={this.signUpWithFacebookAsync}
                      onLongPress={this.signUpWithFacebookAsync} />
                  <Text onPress={this.props.changeFormType} style={styles.text}>
                      Have an account? Sign In
          </Text>
              </View>
          </View>
      );
  }
}

CredentialInputScreen.propTypes = {
  isLoggedIn: PropTypes.bool,
  onSubmit: PropTypes.func,
  changeFormType: PropTypes.func,
  register: PropTypes.func,
  onLogIn: PropTypes.func,
  onLogOut: PropTypes.func,
  navigation: PropTypes.shape({
    navigate: PropTypes.func,
    push: PropTypes.func,
  }),
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    backgroundColor: colors.lightBlue,
    paddingTop: 10,
    paddingBottom: 10,
  },
  inputContainer: {
    backgroundColor: colors.white,
    borderTopLeftRadius: 25,
    borderTopRightRadius: 25,
    borderBottomLeftRadius: 25,
    borderBottomRightRadius: 25,
    flexDirection: 'column',
    justifyContent: 'center',
    marginLeft: 10,
    marginRight: 10,
    maxHeight: 507,
    maxWidth: 369,
    paddingTop: 20,
    paddingBottom: 20,
    paddingLeft: 20,
    paddingRight: 20,
  },
  social: {
    marginBottom: 8,
    marginLeft: 0,
    marginRight: 0,
    alignSelf: 'center',
    width: '100%',
    maxWidth: 322,
  },
  submitButton: {
    textAlign: 'right',
    paddingTop: 20,
    fontSize: 18,
    color: colors.darkBlue,
  },
  textInput: {
    color: colors.black,
    fontSize: 18,
    paddingTop: 10,
    paddingBottom: 10,
    borderBottomWidth: 1,
    borderBottomColor: colors.lightGray2,
  },
  titleText: {
    textAlign: 'left',
    fontSize: 24,
    fontWeight: '500',
    marginBottom: 20,
  },
  text: {
    textAlign: 'center',
    fontSize: 18,
    paddingTop: 10,
    paddingBottom: 20,
  },
});

export default connect(
  (state) => ({ isLoggedIn: getIsLoggedIn(state) }),
  { register, onLogIn: loginSuccess, onLogOut: logOutSuccess },
)(CredentialInputScreen);
