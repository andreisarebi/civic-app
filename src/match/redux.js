import { combineReducers } from 'redux';
import { UserActionType } from '../user/sagas';

// Selectors
export const MATCH_NAMESPACE = 'match';
const SURVEY_NAMESPACE = 'survey';
const POSITIONS_NAMESPACE = 'userPositions';

export const getSurveyQuestions = (state) => (
  state[MATCH_NAMESPACE][SURVEY_NAMESPACE]
);

export const getUserPositions = (state) => (
  state[MATCH_NAMESPACE][POSITIONS_NAMESPACE]
);

// Action Creators
export const loadSurveySuccess = surveyQuestions => ({
  type: MatchActionType.SurveyRequestSuccess,
  payload: surveyQuestions,
});

export const loadSurvey = () => ({
  type: MatchActionType.SurveyRequest,
});

export const MatchActionType = {
  SurveyRequest: 'civicApp/match/survey/REQUEST',
  SurveyRequestSuccess: 'civicApp/match/survey/REQUEST_SUCCESS',
  UserPositionUpdate: 'civicApp/match/positions/UPDATE',
};

// Reducers
const surveyReducer = (state = null, action) => {
  switch (action.type) {
    case MatchActionType.SurveyRequestSuccess:
      return {
        ...state,
        ...action.payload
      };
    default:
      return state;
  }
};

const positionsReducer = (state = {}, action) => {
  switch (action.type) {
    case UserActionType.RequestSuccess:
      return {
        ...state,
        ...action.payload.responses,
      };
    default:
      return state;
  }
};

export default combineReducers({
  [SURVEY_NAMESPACE]: surveyReducer,
  [POSITIONS_NAMESPACE]: positionsReducer,
});
