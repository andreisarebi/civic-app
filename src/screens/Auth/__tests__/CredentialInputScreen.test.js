import { createSnapshot } from '../../../util/tests';
import CredentialInputScreen from '../CredentialInputScreen';
import { formTypes } from '../../../auth/redux';

describe('CredentialInputScreen', () => {
  it('renders correctly in login mode', () => {
    const props = {
      isLoggedIn: false,
      navigation: { navigate: jest.fn() },
      formType: formTypes.LOGIN,
      onSubmit: jest.fn(),
      changeFormType: jest.fn(),
      formIsValid: true,
      email: 'foo@bar.com',
      updateEmail: jest.fn(),
      password: 'password',
      updatePassword: jest.fn(),
      isLoading: false,
      errorMessage: '',
      updateErrorVisibility: jest.fn(),
      showErrors: false,
      duplicatePassword: 'password',
      updateDuplicatePassword: jest.fn(),
    };
    const snapshot = createSnapshot(CredentialInputScreen, props);
    expect(snapshot).toMatchSnapshot();
  });
  it('renders correctly in sign up mode', () => {
    const props = {
      isLoggedIn: false,
      navigation: { navigate: jest.fn() },
      formType: formTypes.SIGN_UP,
      onSubmit: jest.fn(),
      changeFormType: jest.fn(),
      formIsValid: true,
      email: 'foo@bar.com',
      updateEmail: jest.fn(),
      password: 'password',
      updatePassword: jest.fn(),
      isLoading: false,
      errorMessage: '',
      updateErrorVisibility: jest.fn(),
      showErrors: false,
      duplicatePassword: 'password',
      updateDuplicatePassword: jest.fn(),
    };
    const snapshot = createSnapshot(CredentialInputScreen, props);
    expect(snapshot).toMatchSnapshot();
  });
  it('renders correctly with form error', () => {
    const props = {
      isLoggedIn: false,
      navigation: { navigate: jest.fn() },
      formType: formTypes.LOGIN,
      onSubmit: jest.fn(),
      changeFormType: jest.fn(),
      formIsValid: false,
      email: '',
      updateEmail: jest.fn(),
      password: 'password',
      updatePassword: jest.fn(),
      isLoading: false,
      errorMessage: 'Email required',
      updateErrorVisibility: jest.fn(),
      showErrors: true,
    };
    const snapshot = createSnapshot(CredentialInputScreen, props);
    expect(snapshot).toMatchSnapshot();
  });
  it('renders correctly when loading', () => {
    const props = {
      isLoggedIn: false,
      navigation: { navigate: jest.fn() },
      formType: formTypes.LOGIN,
      onSubmit: jest.fn(),
      changeFormType: jest.fn(),
      formIsValid: true,
      email: 'email@email.com',
      updateEmail: jest.fn(),
      password: 'password',
      updatePassword: jest.fn(),
      isLoading: true,
      errorMessage: '',
      updateErrorVisibility: jest.fn(),
      showErrors: false,
    };
    const snapshot = createSnapshot(CredentialInputScreen, props);
    expect(snapshot).toMatchSnapshot();
  });
});
