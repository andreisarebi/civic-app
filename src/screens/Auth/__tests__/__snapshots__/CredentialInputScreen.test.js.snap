// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CredentialInputScreen renders correctly in login mode 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#2d9cdb",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingBottom": 10,
      "paddingTop": 10,
    }
  }
>
  <Component
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderBottomLeftRadius": 25,
        "borderBottomRightRadius": 25,
        "borderTopLeftRadius": 25,
        "borderTopRightRadius": 25,
        "flexDirection": "column",
        "justifyContent": "center",
        "marginLeft": 10,
        "marginRight": 10,
        "maxHeight": 530,
        "maxWidth": 369,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "500",
          "marginBottom": 20,
          "textAlign": "left",
        }
      }
    >
      Sign in
    </Text>
    <EmailInput
      onChangeText={[MockFunction]}
      placeholder={undefined}
      value="foo@bar.com"
    />
    <PasswordInput
      onChangeText={[MockFunction]}
      placeholder={undefined}
      value="password"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "color": "#126390",
          "fontSize": 18,
          "paddingTop": 20,
          "textAlign": "right",
        }
      }
    >
      SUBMIT
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "paddingBottom": 20,
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
    >
      or
    </Text>
    <Connect(GoogleAuthButton)
      style={
        Object {
          "alignSelf": "center",
          "marginBottom": 8,
          "marginLeft": 0,
          "marginRight": 0,
          "maxWidth": 322,
          "width": "100%",
        }
      }
      title="Sign in with Google"
    />
    <Connect(FacebookAuthButton)
      style={
        Object {
          "alignSelf": "center",
          "marginBottom": 8,
          "marginLeft": 0,
          "marginRight": 0,
          "maxWidth": 322,
          "width": "100%",
        }
      }
      title="Continue with Facebook"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "fontSize": 18,
          "paddingBottom": 20,
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
    >
      Don't have an account yet? Register
    </Text>
  </Component>
</Component>
`;

exports[`CredentialInputScreen renders correctly in sign up mode 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#2d9cdb",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingBottom": 10,
      "paddingTop": 10,
    }
  }
>
  <Component
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderBottomLeftRadius": 25,
        "borderBottomRightRadius": 25,
        "borderTopLeftRadius": 25,
        "borderTopRightRadius": 25,
        "flexDirection": "column",
        "justifyContent": "center",
        "marginLeft": 10,
        "marginRight": 10,
        "maxHeight": 530,
        "maxWidth": 369,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "500",
          "marginBottom": 20,
          "textAlign": "left",
        }
      }
    >
      Sign Up for Civic
    </Text>
    <EmailInput
      onChangeText={[MockFunction]}
      placeholder={undefined}
      value="foo@bar.com"
    />
    <PasswordInput
      onChangeText={[MockFunction]}
      placeholder={undefined}
      value="password"
    />
    <PasswordInput
      onChangeText={[MockFunction]}
      placeholder="Re-type password"
      value="password"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "color": "#126390",
          "fontSize": 18,
          "paddingTop": 20,
          "textAlign": "right",
        }
      }
    >
      SUBMIT
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "paddingBottom": 20,
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
    >
      or
    </Text>
    <Connect(GoogleAuthButton)
      style={
        Object {
          "alignSelf": "center",
          "marginBottom": 8,
          "marginLeft": 0,
          "marginRight": 0,
          "maxWidth": 322,
          "width": "100%",
        }
      }
      title="Sign up with Google"
    />
    <Connect(FacebookAuthButton)
      style={
        Object {
          "alignSelf": "center",
          "marginBottom": 8,
          "marginLeft": 0,
          "marginRight": 0,
          "maxWidth": 322,
          "width": "100%",
        }
      }
      title="Continue with Facebook"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "fontSize": 18,
          "paddingBottom": 20,
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
    >
      Have an account? Sign in
    </Text>
  </Component>
</Component>
`;

exports[`CredentialInputScreen renders correctly when loading 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#2d9cdb",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingBottom": 10,
      "paddingTop": 10,
    }
  }
>
  <Component
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderBottomLeftRadius": 25,
        "borderBottomRightRadius": 25,
        "borderTopLeftRadius": 25,
        "borderTopRightRadius": 25,
        "flexDirection": "column",
        "justifyContent": "center",
        "marginLeft": 10,
        "marginRight": 10,
        "maxHeight": 530,
        "maxWidth": 369,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "500",
          "marginBottom": 20,
          "textAlign": "left",
        }
      }
    >
      Sign in
    </Text>
    <EmailInput
      onChangeText={[MockFunction]}
      placeholder={undefined}
      value="email@email.com"
    />
    <PasswordInput
      onChangeText={[MockFunction]}
      placeholder={undefined}
      value="password"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "color": "#126390",
          "fontSize": 18,
          "paddingTop": 20,
          "textAlign": "right",
        }
      }
    >
      SUBMIT
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "paddingBottom": 20,
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
    >
      or
    </Text>
    <Connect(GoogleAuthButton)
      style={
        Object {
          "alignSelf": "center",
          "marginBottom": 8,
          "marginLeft": 0,
          "marginRight": 0,
          "maxWidth": 322,
          "width": "100%",
        }
      }
      title="Sign in with Google"
    />
    <Connect(FacebookAuthButton)
      style={
        Object {
          "alignSelf": "center",
          "marginBottom": 8,
          "marginLeft": 0,
          "marginRight": 0,
          "maxWidth": 322,
          "width": "100%",
        }
      }
      title="Continue with Facebook"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "fontSize": 18,
          "paddingBottom": 20,
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
    >
      Don't have an account yet? Register
    </Text>
  </Component>
</Component>
`;

exports[`CredentialInputScreen renders correctly with form error 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#2d9cdb",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingBottom": 10,
      "paddingTop": 10,
    }
  }
>
  <Component
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderBottomLeftRadius": 25,
        "borderBottomRightRadius": 25,
        "borderTopLeftRadius": 25,
        "borderTopRightRadius": 25,
        "flexDirection": "column",
        "justifyContent": "center",
        "marginLeft": 10,
        "marginRight": 10,
        "maxHeight": 530,
        "maxWidth": 369,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "500",
          "marginBottom": 20,
          "textAlign": "left",
        }
      }
    >
      Sign in
    </Text>
    <EmailInput
      onChangeText={[MockFunction]}
      placeholder={undefined}
      value=""
    />
    <PasswordInput
      onChangeText={[MockFunction]}
      placeholder={undefined}
      value="password"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#eb5757",
          "fontSize": 16,
          "paddingBottom": 4,
          "paddingTop": 4,
          "textAlign": "center",
        }
      }
    >
      Email required
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "color": "#126390",
          "fontSize": 18,
          "paddingTop": 20,
          "textAlign": "right",
        }
      }
    >
      SUBMIT
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "paddingBottom": 20,
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
    >
      or
    </Text>
    <Connect(GoogleAuthButton)
      style={
        Object {
          "alignSelf": "center",
          "marginBottom": 8,
          "marginLeft": 0,
          "marginRight": 0,
          "maxWidth": 322,
          "width": "100%",
        }
      }
      title="Sign in with Google"
    />
    <Connect(FacebookAuthButton)
      style={
        Object {
          "alignSelf": "center",
          "marginBottom": 8,
          "marginLeft": 0,
          "marginRight": 0,
          "maxWidth": 322,
          "width": "100%",
        }
      }
      title="Continue with Facebook"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "fontSize": 18,
          "paddingBottom": 20,
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
    >
      Don't have an account yet? Register
    </Text>
  </Component>
</Component>
`;
